test/framework/filetools.py

```diff
@@ -694,6 +694,22 @@ def test_derive_alt_pypi_url(self):
         url = 'https://pypi.python.org/packages/source/n/nosuchpackageonpypiever/nosuchpackageonpypiever-0.0.0.tar.gz'
         self.assertEqual(ft.derive_alt_pypi_url(url), None)
 
+    def test_apply_patch(self):
+        """ Test apply_patch """
+        testdir = os.path.dirname(os.path.abspath(__file__))
+        tmpdir = tempfile.mkdtemp()
```

Comment:
use `self.test_prefix` (which is cleaned up automagically after the test runs, whether it succeeds or not)